pilot/pkg/model/push_context.go

```diff
@@ -81,6 +82,9 @@ type PushContext struct {
 	ServicePort2Name map[string]PortList `json:"-"`
 
 	initDone bool
+
+	// MixerPerRouteFilterConfig has mixer Router Filter config, indexed by service hostname + outbound/inbound
+	MixerPerRouteFilterConfig map[string]*types.Struct `json:"-"`
```

Comment:
My comment didn't make it through. But can you please send a follow up PR to remove this and move it into the plugins? There is no point in spilling plugin state back into the core leading to all the same issues we had in 0.7.

Reply:
Add a precompute callback to all the plugins.. during initialization, call configgen.BuildSharedState which in turn should call all the precompute callbacks and cache the state in the config gen.